File: demo/src/main/java/com/example/demo/services/TutorService.java
package com.example.demo.services;

import com.example.demo.model.entities.Tutor;


import java.util.List;
import java.util.Optional;


public interface TutorService {
    Tutor saveTutor(Tutor tutor);
    Optional<Tutor> getTutorById(Long id);
    List<Tutor> getAllTutores();
    Tutor updateTutor(Tutor tutor);
    void deleteTutor(Long id);
}

File: escuela/src/app/services/tutor/tutor.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

import { Observable } from 'rxjs';
import { tutor } from 'src/app/models/tutor.interface';
@Injectable({
  providedIn: 'root',
})
export class TutorService {
  private apiUrl = 'http://localhost:8086/api/tutores';

  constructor(private http: HttpClient) {}

  getAllTutores(): Observable<tutor[]> {
    return this.http.get<tutor[]>(this.apiUrl);
  }

  getTutorById(id: number): Observable<tutor> {
    return this.http.get<tutor>(`${this.apiUrl}/${id}`);
  }

  createTutor(tutor: tutor): Observable<tutor> {
    return this.http.post<tutor>(`${this.apiUrl}/create`, tutor);
  }

  updateTutor(tutor: tutor): Observable<tutor> {
    return this.http.put<tutor>(`${this.apiUrl}/${tutor.id}`, tutor);
  }

  deleteTutor(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}